src/zoid/buttons/util.js

```diff
@@ -74,3 +74,38 @@ export function getVenmoExperiment(experiment : ?Experiment) : VenmoExperiment {
         enableVenmo: Boolean(isExperimentEnabled || isEnableFundingVenmo)
     };
 }
+
+export function applePaySession() : ?ApplePaySessionConfigRequest {
+    try {
+        if (!window.ApplePaySession) {
+            return;
+        }
+
+        return (version, request) => {
+            const session = new window.ApplePaySession(version, request);
+            return {
+                addEventListener: (name, handler) => {
+                    session.addEventListener(name, handler);
+                },
+                completeMerchantValidation: (validatedSession) => {
+                    session.completeMerchantValidation(validatedSession);
```

Comment:
Do these return promises? Should we return those so the caller can await them?

Reply:
@bluepnume No.  They are fire and forget and return void: https://developer.apple.com/documentation/apple_pay_on_the_web/applepaysession/1778012-completepayment .  We do have 30s to fire from getting the `onpaymentauthorized` event or they will invalidate the payment which is good if something happens when calling onApprove.